packages/ember-htmlbars/lib/keywords/get.js

```diff
@@ -22,17 +21,90 @@ function labelFor(source, key) {
   return `(get ${sourceLabel} ${keyLabel})`;
 }
 
+let DynamicKeyStream = BasicStream.extend({
+  init(source, keySource) {
+    assert('DynamicKeyStream error: source must be a stream', isStream(source)); // TODO: This isn't necessary.
+
+    // used to get the original path for debugging and legacy purposes
+    var label = labelFor(source, keySource);
+
```

Comment:
No, inside `BasicStream.extend` we create a class that calls its own internal `_init` before this `init` callback.  If we extended something other than `BasicStream` we would need to call super.